src/storage/IndexBaseProcessor.inl

```diff
@@ -0,0 +1,188 @@
+
+
+namespace nebula {
+namespace storage {
+
+template<typename RESP>
+void IndexBaseProcessor<RESP>::doIndexCreate(PartitionID partId) {
+    std::string key, val;
+    StatusOr<std::string> indexKey;
+    std::vector<kvstore::KV> data;
+
+    int32_t batchPutNum = 0;
+    cpp2::ResultCode thriftResult;
+    thriftResult.set_code(cpp2::ErrorCode::SUCCEEDED);
+    thriftResult.set_part_id(partId);
+
+    auto prefix = NebulaKeyUtils::partPrefix(partId);
+    std::unique_ptr<kvstore::KVIterator> iter;
+    auto ret = this->kvstore_->prefixSnapshot(spaceId_, partId, prefix, &iter);
+
+    if (ret != kvstore::ResultCode::SUCCEEDED || !iter) {
+        thriftResult.set_code(this->to(ret));
+        INDEX_CREATE_DONE;
+    }
+
+    while (iter->valid()) {
+        key = iter->key().str();
+        val = iter->val().str();
+        iter->next();
+        indexKey = (indexType_ == nebula::cpp2::IndexType::EDGE) ?
+                   assembleEdgeIndexKey(spaceId_, partId, key, std::move(val)) :
+                   assembleVertexIndexKey(spaceId_, partId, key, std::move(val));
+
+        if (!indexKey.ok()) {
+            continue;
+        }
+        data.emplace_back(std::move(indexKey.value()), std::move(key));
+        batchPutNum++;
+        if (batchPutNum == FLAGS_bulk_number_per_index_creation) {
+            auto code = doBatchPut(spaceId_, partId, data);
+            if (code == cpp2::ErrorCode::E_LEADER_CHANGED) {
+                nebula::cpp2::HostAddr leader;
+                auto addrRet = this->kvstore_->partLeader(spaceId_, partId);
+                CHECK(ok(addrRet));
+                auto addr = value(std::move(addrRet));
+                leader.set_ip(addr.first);
+                leader.set_port(addr.second);
+                thriftResult.set_leader(leader);
+            } else if (code != cpp2::ErrorCode::SUCCEEDED) {
+                thriftResult.set_code(code);
+                INDEX_CREATE_DONE;
+            }
+            data.clear();
+            batchPutNum = 0;
+        }
+    }
+
+    if (!data.empty()) {
+        auto code = doBatchPut(spaceId_, partId, data);
+        if (code == cpp2::ErrorCode::E_LEADER_CHANGED) {
+            nebula::cpp2::HostAddr leader;
+            auto addrRet = this->kvstore_->partLeader(spaceId_, partId);
+            CHECK(ok(addrRet));
+            auto addr = value(std::move(addrRet));
+            leader.set_ip(addr.first);
+            leader.set_port(addr.second);
+            thriftResult.set_leader(leader);
+        } else if (code != cpp2::ErrorCode::SUCCEEDED) {
+            thriftResult.set_code(code);
+            INDEX_CREATE_DONE;
+        }
+        data.clear();
+    }
+
+    // TODO : send index status 'CONSTRUCTING' via meta client,
+    //        should block all actions of insert|delete| update.
+    //        until index create done of each parts.
+
+    // TODO : Assemble new data changes during index creation.
+
+    INDEX_CREATE_DONE;
```

Comment:
the micro will let gdb step by step hard, and let the code understand hard, change to function like finishExecution?

Reply:
That sounds good.